content/tutorial-react-native/rn-03.md

```diff
@@ -0,0 +1,343 @@
+    const ds = new ListView.DataSource({
+      rowHasChanged: (r1, r2) => r1.id !== r2.id
+    });
+    this.state = {
+      dataSource: ds.cloneWithRows([]),
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { data } = nextProps;
+    // Only change the rows if some data is comming
+    if (!data.loading && !data.error) {
+      const { dataSource } = this.state;
+      this.setState({
+        dataSource: dataSource.cloneWithRows(data.Trainer.ownedPokemons),
+      });
+    }
+  }
+
+  render() {
+    if (this.props.data.error) {
+      // return error text
+    }
+
+    if (this.props.data.loading) {
+      // return ActivityIndicator loader
+    }
+
+    return (
+      <ListView
+        enableEmptySections
+        dataSource={this.state.dataSource} // the datastructure
+        renderRow={this._renderItem}
+        renderHeader={this._renderHeader}
+        renderScrollComponent={props => <RecyclerScrollView {...props} />}
+        renderSeparator={this._renderSeparator}
+        style={styles.listView}
+        pageSize={1}
+        initialListSize={10}
+      />
+    );
+  }
+
+  _renderItem = (pokemon) => (
+    <PokemonListItem pokemon={pokemon} />
+  );
+
+  _renderHeader = (trainer) => (
+    <Title trainer={this.props.data.Trainer} />
+  );
+
+  _renderSeparator = (sectionID, rowID) => (
+    <View
+      key={`${sectionID}-${rowID}`}
+      style={{ height: 1, backgroundColor: '#CCCCCC'}}
+    />
+  );
+}
+```
+
+You can then modify `Title` to accept `this.props;data.Trainer` as props `trainer` and use of 
+`this.props.trainer.ownedPokemons.length` that displays the correct amount of pokemons.
+
+To confirm your environment is all correctly setup, start the app now from XDE.
+
+
+### Dynamic Query Variables
+
+If everything working, we can now continue to implement a detailed view of a single pokemon when we click on a 
+`PokemonListItem` component. 
+
+Have a look at the `PokemonDetailScreen` directory that we prepared for you. 
+The component in `screen/PokemonDetailScreen/components/PokemonDetail.js` render the screen based on the props pokemon 
+that the page pass.
+
+We could have use the existing data from the list to rendre the whole screen, without an extra request but we will still
+do the extra request to show yout how to customize the request based on the route.
+
+We already created the navigation in `navigation/` that assigns the `PokemonDetailScreen` to the path `pokemonDetail` 
+so we can. Exponent navigation allows us to pass extra parameters to the route like this :
+
+```js
+_goToDetail = (pokemon) => {
+  this.props.navigation
+    .getNavigator('root')
+    .push(Router.getRoute('pokemonDetail', { pokemon }));
+}
+```
+
+where `getNavigator` allows to select the root navigator (and not the tab navigator) and push some routes on it, which makes
+the second order pages'tab bar disappear, and `push` allow to pass an extra object of parameters.
```

Comment:
The idea I want to express is "native app usually have a tab bar only in the first layer. As soon as you click one item in the page, the detail page have no tab bar." But indeed that has no sense.